pkg/storage/stores/tsdb/multi_file_index.go

```diff
@@ -142,11 +134,14 @@ func (i *MultiIndex) GetChunkRefs(ctx context.Context, userID string, from, thro
 		from,
 		through,
 		func(ctx context.Context, idx Index) error {
-			got, err := idx.GetChunkRefs(ctx, userID, from, through, nil, shard, matchers...)
+			var err error
+			buf := ChunkRefsPool.Get()
+			defer ChunkRefsPool.Put(buf)
```

Comment:
I see a problem here. We are adding back buf to the pool while it is still being used in the accumulator. Am I missing something?

Reply:
No, you are right, the buffer must only be put back once it has been merged.